### PR TITLE
Changing build_and_test.sh script to include tests for region spannin…

### PR DESCRIPTION
**Pull request by gautham-bhat-k**
…g and multiple memory server configuration

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -116,6 +116,7 @@ include_directories(${PROJECT_SOURCE_DIR}/src/pmi) # private headers (src)
 # library directories
 #
 link_directories(${PROJECT_SOURCE_DIR}/third-party/build/lib) # third-party libraries
+link_directories(${PROJECT_SOURCE_DIR}/third-party/build/lib64) # third-party libraries
```
Comment (mashoodk) [inline CMakeLists.txt:119]: Why is this change necessary ? What is it getting installed in lib64 ?

File: build_and_test.sh [hunk 2]
```diff
@@ -92,4 +92,18 @@ then
         echo "OpenFAM test with shared memory configuration failed. exit..."
         exit 1
 fi
+
```
Comment (mashoodk) [inline build_and_test.sh:95]: I am thinking if we should keep it as a separate script and not to run always we run build_and_test.sh. 
What do you think @faizan-barmawer.

File: scripts/run_test.py [hunk 3]
```diff
@@ -6,357 +6,559 @@
...
+    os.environ["OPENFAM_TEST_OPT"] = (
+        "-N "
+        + str(len(args.pehosts.split(",")))
+        + " -n "
+        + str(npe)
+        + " --nodelist="
+        + args.pehosts
+        + " --mpi=pmix_v2"
+    )
+# Check if user has provided any service interface as "rpc" when openfam model is shared_memeoyr
```
Comment (mashoodk) [inline scripts/run_test.py:271]:
```suggestion
# Check if user has provided any service interface as "rpc" when openfam model is shared_memory
```

File: scripts/run_test.py [hunk 4]
```diff
@@ -6,357 +6,559 @@
...
+                + " \"sh -c '"
+                + env_cmd
+                + "nohup "
+                + args.buildpath
+                + "/src/memory_server -a "
+                + memory_server_addr
+                + " -r "
+                + str(memory_server_rpc_port)
+                + " -l "
+                + str(memservice_config_doc["libfabric_port"])
```
Comment (mashoodk) [inline scripts/run_test.py:360]: Are we providing same libfabric port for all memory servers ?

**Review by mashoodk — CHANGES_REQUESTED** [review]

**Review by mashoodk — CHANGES_REQUESTED** [review]

**Review by BA-Ramya — CHANGES_REQUESTED** [review]
for slurm , replace pkill commands as shown:
scancel --quiet -n metadata_server  > /dev/null 2>&1
scancel --quiet -n memory_server  > /dev/null 2>&1
scancel --quiet -n cis_server  > /dev/null 2>&1
This is because many clusters may not allow more than one job to run

**Review by mashoodk — APPROVED** [review]

**Review by BA-Ramya — APPROVED** [review]

**Pull request merged**